### ISSUE TITLE
Tiled Gallery: Accessibility - when set to link to 'None', remove button role, and tabIndex of 0

### ISSUE THREAD

**Issue by coder-karen** [Automattic/jetpack#39925]
### Issue

A Tiled Gallery set to link to 'None' defines a role of button and tabIndex of 0. Both clicking on it and tabbing to it then pressing enter result in no action being taken, which is (I would assume) expected. It appears in this case that the role and tabIndex should then be removed.

### How to reproduce

This was tested on self-hosted and WoA.

1. On a test site, ensure the full-screen Carousel option is disabled under Jetpack > Settings > Writing: `/wp-admin/admin.php?page=jetpack#/writing`
2. Create a new post and add a Tiled Gallery block.
3. Ensure 'None' is set for the 'Link to' option in the sidebar:
![Image](https://github.com/user-attachments/assets/b01fe11b-0e6d-44d9-b942-a17b2deba651)

4. Save and preview
5. If you inspect the image source for one of the Tiled Gallery images, you'll see a tabIndex of 0, and role of button:

![Image](https://github.com/user-attachments/assets/fd8c70b2-b3d2-4376-8b9b-1b2ceeb7ca28)

6. Click on any of the images - you'll see the image highlighted but no action taken
7. Similarly, tab onto one of the images and press enter. You'll see the image highlighted, but no action taken.

### What should happen

I would assume that since no action should be taken, the role and tabIndex should not be set as they are.

### Related conversations

https://github.com/Automattic/jetpack/pull/39913#pullrequestreview-2398585938
